curve25519-dalek/src/lizard/lizard_ristretto.rs

```diff
@@ -93,11 +91,15 @@ impl RistrettoPoint {
 
         let jcs = self.to_jacobi_quartic_ristretto();
 
-        // Compute the inverse of the every point and its dual
-        let invs = jcs.iter().flat_map(|jc| [jc.e_inv(), jc.dual().e_inv()]);
+        // Compute the positive solution to e⁻¹ on every point and its dual
+        let pos_invs = jcs
+            .iter()
+            .flat_map(|jc| [jc.e_inv_positive(), jc.dual().e_inv_positive()]);
+        // Compute the other solutions to e⁻¹, ie the negatives of the above solutions
+        let neg_invs = pos_invs.clone().map(|mx| mx.map(|x| -&x));
```

Comment:
This will repeat work, since Rust iterators are lazy. Better to collect it in an Array first and then do the negation in a second round.